### PR TITLE
cnap/core: add key broker module to get the key from KBS

### PR DESCRIPTION
**Pull request by dongx1x**
Add a KeyBrokerClientBase abstract class and add a simple implementation class SimpleKeyBrokerClient to show how to get a key from the key broker server.

### CODE REVIEW

File: cnap/requirements.txt [hunk 1]
```diff
@@ -9,3 +9,4 @@ redis
 tensorflow
 websockets
 requests
+ccnp
```
Comment (leyao-daily) [inline cnap/requirements.txt]: Make the requirements sorted.

File: cnap/core/keybroker.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        headers = {
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+            "Attestation-Type": "TDX"
+        }
+
+        LOG.debug("Getting key from the KBS")
+        # retry 3 times for the KBS server issue
+        resp = None
+        for _ in range(3):
```
Comment (Hulongyin) [inline cnap/core/keybroker.py]: Maybe a `RETRY_TIMES=3` will be more clear?

File: cnap/core/keybroker.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+        raise NotImplementedError("Subclasses should implement get_key() method.")
+
+
+class SimpleKeyBrokerClient(KeyBrokerClientBase):
+    """A implementation for key broker client.
+
+    This class implement `get_key` in `KeyBrokerClientBase` abstract base class to connect
+    to a simple KBS.
+    Here is an example flow of a simple KBS:
+        * Accept a quote and a public key from client.
```
Comment (hwang37) [inline cnap/core/keybroker.py]: It's better to use Seq# to replace "*".
Same comment to the follow description below.

File: cnap/core/keybroker.py [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+        }
+
+        LOG.debug("Getting key from the KBS")
+        # retry 3 times for the KBS server issue
+        resp = None
+        for _ in range(3):
+            try:
+                resp = requests.post(server_url, json=req_body, headers=headers, verify=False,
+                                     timeout=TIMEOUT)
+                if resp.status_code in [200]:
```
Comment (hwang37) [inline cnap/core/keybroker.py]: magic number 200 here and below, should we define it as MACRO (term in C language)?

File: cnap/requirements.txt (hunk 1 above)
Comment (dongx1x) [reply cnap/requirements.txt]: Done.

File: cnap/core/keybroker.py (hunk 2 above)
Comment (dongx1x) [reply cnap/core/keybroker.py]: Done.

File: cnap/core/keybroker.py (hunk 3 above)
Comment (dongx1x) [reply cnap/core/keybroker.py]: How about changing it to `-`, it should be generated as a bullet point list in docstring.

File: cnap/core/keybroker.py (hunk 4 above)
Comment (dongx1x) [reply cnap/core/keybroker.py]: Done.

**Review by Hulongyin — APPROVED** [review]
LGTM. Thanks

**Pull request merged**